modules/reactor/src/meshgenerators/PatternedCartesianMeshGenerator.C

```diff
@@ -0,0 +1,752 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PatternedCartesianMeshGenerator.h"
+#include "MooseUtils.h"
+#include "MooseMeshUtils.h"
+
+// C++ includes
+#include <cmath> // provides round, not std::round (see http://www.cplusplus.com/reference/cmath/round/)
+#include <fstream> // used to generate the optional control drum position file
+
+registerMooseObject("ReactorApp", PatternedCartesianMeshGenerator);
+
+InputParameters
+PatternedCartesianMeshGenerator::validParams()
+{
+  InputParameters params = PolygonMeshGeneratorBase::validParams();
+  params.addRequiredParam<std::vector<MeshGeneratorName>>("inputs", "The input MeshGenerators.");
+  params.addRequiredRangeCheckedParam<std::vector<std::vector<unsigned int>>>(
+      "pattern",
+      "pattern>=0",
+      "A double-indexed square-shaped array starting with the upper-left corner.");
+  MooseEnum cartesian_pattern_boundary("none expanded", "expanded");
```

Comment:
oh the docs calls it 'square', we need to be consistent?

Reply:
Will also mention cartesian in the doc

Reply:
It should be `expanded`